tests/test_kernels.py

```diff
@@ -26,18 +27,193 @@ def grid_type_field_and_extra_kwargs(request):
+            test_kwargs["dxn"][random_yloc + 1, :] = 2000
+
+        LaplacianClass = ALL_KERNELS[grid_type]
+        laplacian = LaplacianClass(**test_kwargs)
+        diffused = laplacian(delta)
+        # check that delta function gets diffused isotropically in y-direction
+        np.testing.assert_allclose(
+            diffused[random_yloc - 1, random_xloc],
+            diffused[random_yloc + 1, random_xloc],
+            atol=1e-12,
+        )
+
+
+def test_flux_in_x_direction(grid_type_field_and_extra_kwargs):
+    """This test checks that the Laplacian computes the correct fluxes in x-direction if the grid is irregular.
+    The test will catch sign errors in the Laplacian rolling of array elements along the x-axis."""
+    grid_type, data, extra_kwargs = grid_type_field_and_extra_kwargs
+
+    if grid_type in irregular_grids:
+        # deploy mass at random location away from Antarctica: delta_{j,i}
+        delta = np.zeros_like(data)
+        ny = np.shape(delta)[0]
+        random_yloc = np.random.randint(5, ny)
+        nx = np.shape(delta)[1]
+        random_xloc = np.random.randint(0, nx)
+        delta[random_yloc, random_xloc] = 1
+
+        test_kwargs = copy.deepcopy(extra_kwargs)
+        ones = np.ones_like(data)
+        # create highly irregular grid data for
+        # - western edge of cell {j,i-1};
+        # - western edge of cell {j,i+2}
+        # this should leave laplacian(delta_{j,i}) unaffected
+        if grid_type == GridType.IRREGULAR_WITH_LAND:
+            test_kwargs["wet_mask"] = ones  # no land for simplicity
+            test_kwargs["area"] = ones
+            test_kwargs["dys"] = ones
+            test_kwargs["dxw"] = ones
+            test_kwargs["dxs"] = ones
+            test_kwargs["dyw"] = ones.copy()
+            test_kwargs["dyw"][:, random_xloc - 1] = 1000
+            test_kwargs["dyw"][:, random_xloc + 2] = 2000
+        if grid_type == GridType.TRIPOLAR_POP_WITH_LAND:
+            test_kwargs["wet_mask"] = ones.copy()
+            test_kwargs["wet_mask"][0, :] = 0  # Antarctica
+            test_kwargs["tarea"] = ones
+            test_kwargs["dyn"] = ones
+            test_kwargs["dxe"] = ones
+            test_kwargs["dxn"] = ones
+            test_kwargs["dye"] = ones.copy()
+            test_kwargs["dye"][:, random_xloc - 2] = 1000
+            test_kwargs["dye"][:, random_xloc + 1] = 2000
+
+        LaplacianClass = ALL_KERNELS[grid_type]
+        laplacian = LaplacianClass(**test_kwargs)
+        diffused = laplacian(delta)
+        # check that delta function gets diffused isotropically in x-direction
+        np.testing.assert_allclose(
+            diffused[random_yloc, random_xloc - 1],
+            diffused[random_yloc, random_xloc + 1],
+            atol=1e-12,
+        )
+
+
+################## Tripolar grid tests ##############################################
+tripolar_grids = [
+    member
+    for name, member in GridType.__members__.items()
```

Comment:
Why do you need to use `GridType.__members__.items()` here? Does `for name in GridType` not work?

Reply:
`for name in GridType` raises the error:
```
TypeError: argument of type 'GridType' is not iterable
```

Reply:
That is puzzling to me. `GridType` is an enum. The following works.

```
import enum
GridType = enum.Enum("GRID_TYPE", ["A", "B"])
assert [gt.name for gt in GridType] == ["A", 'B']
```